### DISCUSSION TITLE
Inquiry about SAME_DIRECTION_ONLY Functionality in OpenTCS

### DISCUSSION THREAD

**Discussion by BenYaho** [openTCS/opentcs#100] [General]
Hi,
I hope this email finds you well. I am writing to inquire about the "SAME_DIRECTION_ONLY" functionality within the OpenTCS system. During testing on the map provided, I observed that despite enabling the "SAME_DIRECTION_ONLY" option within a block, the system seemed to behave as if the "SINGLE_VEHICLE_ONLY" option was active instead.

I am seeking clarification on how the "SAME_DIRECTION_ONLY" mechanism determines the direction and whether there are specific criteria it follows to ensure vehicles move in the same direction within a block. It would greatly assist me to have a more comprehensive understanding of this feature.

Vehicle-0001   Point-0011 => Point-0020
Vehicle-0002   Point-0018 => Point-0017
Vehicle-0003   Point-0016 => Point-0012

![image](https://github.com/openTCS/opentcs/assets/128113709/79d19c43-d2f9-4ea8-8344-0b715369b137)

**Comment by martingr** [comment]
How the `SAME_DIRECTION_ONLY` block type works is described in the user's guide. Please see the corresponding seciton about blocks ([1]). You have to set the `tcs:blockEntryDirection` property on the paths that lead into the block, but are not themselves part of the block. Paths that correspond to the same entry direction should have their property set to the same value.

[1]: https://www.opentcs.org/docs/5.9/user/opentcs-users-guide.html#_block
